fireant/slicer/widgets/highcharts.py

```diff
@@ -0,0 +1,323 @@
+                                          series_colors,
+                                          groups,
+                                          render_series_label,
+                                          references)
+
+        x_axis = self._render_x_axis(data_frame, dimension_display_values)
+        plot_options = self._render_plot_options(data_frame)
+
+        return {
+            "title": {"text": self.title},
+            "xAxis": x_axis,
+            "yAxis": y_axes,
+            "plotOptions": plot_options,
+            "series": series,
+            "tooltip": {"shared": True, "useHTML": True},
+            "legend": {"useHTML": True},
+        }
+
+
+    def _render_x_axis(self, data_frame, dimension_display_values):
+        """
+        Renders the xAxis configuraiton.
+
+        https://api.highcharts.com/highcharts/yAxis
+
+        :param data_frame:
+        :param dimension_display_values:
+        :return:
+        """
+        first_level = data_frame.index.levels[0] \
+            if isinstance(data_frame.index, pd.MultiIndex) \
+            else data_frame.index
+
+        if isinstance(first_level, pd.DatetimeIndex):
+            return {"type": "datetime"}
+
+        categories = ["All"] \
+            if isinstance(first_level, pd.RangeIndex) \
+            else [utils.deep_get(dimension_display_values,
+                                 [first_level.name, dimension_value],
+                                 dimension_value)
+                  for dimension_value in first_level]
+
+        return {
+            "type": "category",
+            "categories": categories,
+        }
+
+    def _render_y_axis(self, axis_idx, color, references):
+        """
+        Renders the yAxis configuraiton.
+
+        https://api.highcharts.com/highcharts/yAxis
+
+        :param axis_idx:
+        :param color:
+        :param references:
+        :return:
+        """
+        y_axes = [{
+            "id": str(axis_idx),
+            "title": {"text": None},
+            "labels": {"style": {"color": color}}
+        }]
+
+        y_axes += [{
+            "id": "{}_{}".format(axis_idx, reference.key),
+            "title": {"text": reference.label},
+            "opposite": True,
+            "labels": {"style": {"color": color}}
+        }
+            for reference in references
+            if reference.is_delta]
+
+        return y_axes
+
+    def _render_plot_options(self, data_frame):
+        """
+        Renders the plotOptions configuration
+
+        https://api.highcharts.com/highcharts/plotOptions
+
+        :param data_frame:
+        :return:
+        """
+        first_level = data_frame.index.levels[0] \
+            if isinstance(data_frame.index, pd.MultiIndex) \
+            else data_frame.index
+
+        if isinstance(first_level, pd.DatetimeIndex):
+            epoch = np.datetime64(datetime.utcfromtimestamp(0))
+            ms = np.timedelta64(1, 'ms')
+            millis = [d / ms
```

Comment:
Assuming the ms above is milliseconds, I'd rename this to something more meaningful like date_in_milliseconds or date_in_ms.